ctools/defragment_sharded_collection.py

```diff
@@ -82,13 +82,16 @@ async def split_chunk(self, chunk, maxChunkSize_kb):
             }, codec_options=self.cluster.client.codec_options)
 
         if len(res['splitKeys']) > 0:
-            await conn.adminDb.command({
-                    'splitChunk': self.name,
-                    'bounds': [chunk['min'], chunk['max']],
-                    'splitKeys': res['splitKeys']
+            for key in res['splitKeys']:
+                res = await self.cluster.adminDb.command({
+                    'split': self.name,
+                    'middle': key
                 }, codec_options=self.cluster.client.codec_options)
 
-        conn.close()        
+                if res['ok'] != 1.0:
```

Comment:
If result has `ok: 0.0` the pymongo framework will bubble up an exception, so I guess you want to catch the exception here instead of checking the result manually

Reply:
Then I guess this falls under the broader scope of properly catching exceptions, probably not under this PR

Reply:
Yes let's address commands failures in another PR